doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,91 @@
+"""
+============================================================
+Reconstruct dust-covered human cornea image using inpainting
+============================================================
+
+Optical Coherence Tomography (OCT) [1]_ is used to provide eye doctors with an
+image of the retina in the back of a patient's eye.
+
+Quite commonly, there is dust on the reference mirror of the equipment,
+causing dark spots to appear on the acquired images.
+
+This tutorial is adapted from an application shared by Jules Scholler in [2]_.
+The images were acquired by Viacheslav Mazlin.
+
+.. [1] Vinay A. Shah, MD (2015)
+       `Optical Coherence Tomography <https://eyewiki.aao.org/Optical_Coherence_Tomography#:~:text=3%20Limitations-,Overview,at%20least%2010%2D15%20microns.>`_,
+       American Academy of Ophthalmology.
+.. [2] Jules Scholler (2019) "Image denoising using inpainting"
+       https://www.jscholler.com/2019-02-28-remove-dots/
+
+"""
+
+import imageio.v3 as iio
+import numpy as np
+import plotly.io
+import plotly.express as px
+
+from skimage import filters
+
+
+#####################################################################
+# The dataset we are using in this example is an image sequence showing the
+# palisades of Vogt in a human cornea in vivo. Basically, it is a
+# black-and-white movie!
+
+#####################################################################
+# Load image data
+# ===============
+
+image_seq = iio.imread('https://gitlab.com/mkcor/data/-/raw/70eb189f9b1c512fc8926891a2bdf96b67dcf441/in-vivo-cornea-spots.tif')
+
+print(f'number of dimensions: {image_seq.ndim}')
+print(f'shape: {image_seq.shape}')
+print(f'dtype: {image_seq.dtype}')
+
+#####################################################################
+# The dataset is a timeseries of 60 (2D) images. We can visualize it by taking
+# advantage of the `animation_frame` parameter in Plotly's `imshow` function.
+
+fig = px.imshow(
+    image_seq,
+    animation_frame=0,
+    binary_string=True,
+    labels=dict(animation_frame='time point'),
+    title='In-vivo human cornea'
+)
+plotly.io.show(fig)
+
+#####################################################################
+# Average over time
+# =================
+
```

Comment:
We're not "restoring the dust-covered dark spots:" We're restoring the *dust-covered areas/regions* (basically what's under, what's hidden by the dark spots). Also, the logic doesn't work in this first sentence... Again, there's too much of a shortcut. The reason we want to contrast (...) is to threshold, the reason we want to threshold is to segment the dark spots.

Let's tell a story in chronological order! I'd suggest something along the lines of: "First, we want to detect those dark spots where the data are lost."

The overall motivation (reconstructing the whole signal) has already been stated, so we shouldn't repeat it here, as we're about to detail the first processing step.

Reply:
@mkcor I think there was confusion here. All this time I thought that 'dark spots' was referring to the little tiny dots in the image (i.e. the original signal that has been lost). I will reword my explanations, thanks for the clarity!

Reply:
The 'dark spots' are not lost signals but the **dusts**.  
Our task to reconstruct the signal covered by the dark spots. 
To isolate the dark spots with mask configurations coupled with morphological operations, then apply the `inpainting` algorithm. 
The algorithm uses non-damaged pixels around damaged region to gradually reconstruct the lost signal.

O

Reply:
Yeah, I do have a better understanding of the usage of 'dark spots' now. Thanks for the clarification @decorouz ! :)

Reply:
Hey @mkcor ! I have changed the explanation here, does this work?

Reply:
Right, by 'dark spots' we mean the dust-covered regions. We need to *segment* these (verbs like 'reveal' or 'detect' are supposed to mean the same thing in this context, but the technical term would be 'segment').

Reply:
@ana42742 I very much appreciate your effort in terms of clarity, it really shows!

As I wrote in my first comment, I think the first sentence (which you have changed but, semantically, it starts exactly the same way) is out of place. Don't start with "To recover the lost signal in the image sequence," because, again, that's the *overall* goal we are trying to achieve. When you start a section, you should start discussing *directly* what the section *specifically* is about.  It might be relevant to recall the *overall* goal (of the tutorial) at some point in the development of that section, but there's no reason to do so right at the beginning.

> "Basically, we separate out the dark regions from the rest of the image to get the 'true' signal!"

This is wrong: The *separation* per se (segmentation, in technical terms) does not give you the true signal. Where, anyway? The 'separation' is only one step, which gives you a mask, so you decide where to apply inpainting, which in turn is supposed to give you an approximation of the 'true' signal in those regions.  I know that you've understood correctly 'in your head' but writing it down is... a challenge. :pray:

Reply:
> The 'separation' is only one step, which gives you a mask, so you decide where to apply inpainting, which in turn is supposed to give you an approximation of the 'true' signal in those regions. 

@mkcor I get what you mean! How does "...separate out the dark regions from the rest of the image to get a step closer to approximating the 'true' signal!" sound?

Reply:
Suggestion: "in any frame in the sequence" --> "for all frames in the sequence"

Reply:
I've turned this addition into 997fabec99e60187f035f4edc1297cb908c04570.